packages/dds/map/src/directory.ts

```diff
@@ -682,7 +684,8 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
         const stack: [SubDirectory, IDirectoryDataObject][] = [];
         stack.push([this.root, data]);
         while (stack.length > 0) {
-            const [currentSubDir, currentSubDirObject] = stack.pop();
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const [currentSubDir, currentSubDirObject] = stack.pop()!;
```

Comment:
This is a generic question - Is it better to have no-null-assertion or explicitly assert that the value is not null? Whenever I have to add an eslint warning, I kind of assume that it may not be the right way to do it.

Reply:
explicitly assert has runtime impact. in this case, we have check stack.length > 0, so pop can't be undefined. 
So a type assertion is fine in this case, because it is working around the limitation of the lint rule.

Reply:
This is why I love code review; I always learn!